crates/ruff_python_formatter/tests/fixtures.rs

```diff
@@ -277,6 +278,110 @@ fn format() {
     );
 }
 
+/// Tests that the following operations on a file give the same answer:
+/// - First change all f-strings to t-strings and vice-versa, then format.
+/// - First format, then change all f-strings to t-strings and vice-versa
+///
+/// (In other words: formatting is _equivariant_ with respect to interchanging
+/// f and t-strings).
+#[test]
+fn f_and_t_strings_format_the_same() {
+    let test_black_file = |input_path: &Path| {
+        let options_path = input_path.with_extension("options.json");
+        let options = if let Ok(options_file) = fs::File::open(&options_path) {
+            let reader = BufReader::new(options_file);
+            serde_json::from_reader::<BufReader<_>, PyFormatOptions>(reader)
+                .unwrap_or_else(|_| {
+                    panic!("Expected option file {options_path:?} to be a valid Json file")
+                })
+                .with_target_version(PythonVersion::PY314)
+        } else {
+            PyFormatOptions::from_extension(input_path).with_target_version(PythonVersion::PY314)
+        };
+
+        f_and_t_strings_format_the_same_with_options(input_path, &options);
+    };
+
+    let test_ruff_file = |input_path: &Path| {
+        let options_path = input_path.with_extension("options.json");
+        if let Ok(options_file) = fs::File::open(&options_path) {
+            let reader = BufReader::new(options_file);
+            let options_vec = serde_json::from_reader::<BufReader<_>, Vec<PyFormatOptions>>(reader)
+                .unwrap_or_else(|_| {
+                    panic!("Expected option file {options_path:?} to be a valid Json file")
+                });
+
+            for options in options_vec {
+                f_and_t_strings_format_the_same_with_options(
+                    input_path,
+                    &options.with_target_version(PythonVersion::PY314),
+                );
+            }
+        } else {
+            let options = PyFormatOptions::from_extension(input_path)
+                .with_target_version(PythonVersion::PY314);
+            f_and_t_strings_format_the_same_with_options(input_path, &options);
+        }
+    };
+    insta::glob!(
+        "../resources",
+        "test/fixtures/black/**/*.{py,pyi}",
+        test_black_file
+    );
+    insta::glob!(
+        "../resources",
+        "test/fixtures/ruff/**/*.{py,pyi}",
+        test_ruff_file
+    );
```

Comment:
This seems like a great way to verify your implementation but I've concerns about maintaining this long-term. Maybe we should just remove it before landing your PR.

I don't think we should run this on all test files. It unnecessarily increases the test wall time where most tests don't even use f-strings. 

I'm open to keeping this for a handful of tests but I'm more inclined to simply duplicate some f-string tests and update them to use t-strings. 

This will also be hard to maintain if we ever decide to diverge formatting between f- and t-strings.